cordova-lib/spec-cordova/util.spec.js

```diff
@@ -183,4 +184,138 @@ describe('util module', function() {
             expect(res.indexOf('CVS')).toEqual(-1);
         });
     });
+    describe('getPlatformDetailsFromDir function', function () {
+
+        var dir = 'C:\\Projects\\cordova-projects\\cordova-android';
+        var getPackageJsonContentOriginal = util.__get__('getPackageJsonContent');
+        var resolvePathOriginal = util.__get__('resolvePath');
+        var getPlatformDetailsFromDir = util.__get__('getPlatformDetailsFromDir'); // function under test
+
+        beforeEach(function () {
+            util.__set__('getPackageJsonContent', function (p) {
+                return p + '\\package';
+            });
+            util.__set__('resolvePath', function (p) {
+                return p;
+            });
+        });
+
+        afterEach(function () {
+            util.__set__('getPackageJsonContent', getPackageJsonContentOriginal);
+            util.__set__('resolvePath', resolvePathOriginal);
+        });
+
+        it('returns the appropriate platform details', function (done) {
+
+            // Mock out package.json content
+            util.__set__('getPackageJsonContent', function (p) {
+                return {
+                    'name': 'cordova-android',
+                    'version': '3.7.0-dev',
+                    'description': 'cordova-android release',
+                    'main': 'bin/create'
+                };
+            });
+
+            getPlatformDetailsFromDir(dir).then(function (platformDetails) {
+                expect(platformDetails.platform).toBe('android');
+                expect(platformDetails.libDir).toBe(dir);
+                done();
+            });
+        });
+
+        it('throws if the directory supplied does not contain a package.json file', function (done) {
+
+            util.__set__('getPackageJsonContent', function (p) {
```

Comment:
Better to use Jasmine's spyOn(util, 'getPackageJsonContent'), since jasmine will take care of restoring the original function after each spec